botocore/handlers.py

```diff
@@ -109,6 +110,24 @@ def decode_console_output(parsed, **kwargs):
             logger.debug('Error decoding base64', exc_info=True)
 
 
+def generate_idempotent_uuid(params, model, **kwargs):
```

Comment:
It would be great to have a unit test here. Mainly to test that this function properly injects a uuid. You would not have to guess the exact value of the uuid but make sure it looks like an uuid.